### PR TITLE
create test for cleaning up directory for deployment

### PR DESCRIPTION
**Pull request by jarylolz12**
Hello kuya @royge, updates is for the remote-builder integration. As per suggestion we will create function that will remove/cleanup the workspace directory and retain the scripts directory for deployment. This function will do the cleanup based on a condition of:

- if the BUILD_ID is present, that is where the cleanup function work, otherwise it will not remove anything.
- if the scripts path is not provided, it will do nothing and will cause error.

Additionally I added unit test as per suggestion but commented out the unit test for actual `removing of the files`. There is a instruction that must be a done first before executing/commenting out the test function. If you have time, I would like to request for review. Thank you

### CODE REVIEW

**Comment by royge** [comment]
Unsa'y purpose ani `dummy-script` Jar?

**Comment by jarylolz12** [comment]
> Unsa'y purpose ani `dummy-script` Jar?

pag i uncomment nimo ni siya nga [test](https://github.com/royge/shutil/pull/2/commits/08dfc2e7722b5de38e867cbd0e6b14f14d9ab035#diff-4bb157c07a5c7d99c0fdacb42f7c26f7a5785672dbbe39c77f01e70529359201R521) kuya @royge kay gamiton na niya ng `dummy script` nga directory as part sa exception sa pag remove sa mga files. Wala nako g static ug define ang `scripts` nga directory for future changes. Ang purpose lang niya kuya para ma imitate nya ang actual `scripts` directory sa mga repository

**Comment by royge** [comment]
> > Unsa'y purpose ani `dummy-script` Jar?
> 
> pag i uncomment nimo ni siya nga [test](https://github.com/royge/shutil/pull/2/commits/08dfc2e7722b5de38e867cbd0e6b14f14d9ab035#diff-4bb157c07a5c7d99c0fdacb42f7c26f7a5785672dbbe39c77f01e70529359201R521) kuya @royge kay gamiton na niya ng `dummy script` nga directory as part sa exception sa pag remove sa mga files. Wala nako g static ug define ang `scripts` nga directory for future changes. Ang purpose lang niya kuya para ma imitate nya ang actual `scripts` directory sa mga repository

Murag misleading ang contents sa `dummy-scripts` @jarylolz12 kay puede man ma-execute ang scripts inside. Dili nga para testing purposes. Puede ra ni mag-butang ta ug dummy files like `dummy.txt`, `dummy.sh`, etc inside sa `dummy-scripts`.

File: dummy-script/funcs.sh [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+  local app_name=$3
+  local other_file=$4
+
+  if [[ $build_id != "" ]];
+  then
+    [ -z "$script_dir_name" ] && echo "provide the 'scripts' directory" && exit 1;
+    [ -z "$app_name" ] && echo "provide the 'app_name/go-build' file" && exit 1;
+    [ -z "$other_file" ] && echo "See Dockerfile for other files included during docker-build" && exit 1;
+
+    for dir in $(ls); do
```
Comment (royge) [inline dummy-script/funcs.sh:220]: Para safer ni @jarylolz12 puede ta mag-add ug target directory sa `ls` nga command. Kung naa'y target directory puede siya ma-override sa tests.

Example:

```bash
for dir in $(ls $target); do
...
```

File: funcs_test.sh [hunk 2]
```diff
@@ -429,3 +509,13 @@ test_get_version_from_rc
...
+
+test_deployment_cleanup_without_all_parameters
+test_deployment_cleanup_without_build_id
+test_deployment_cleanup_without_script_dir
+test_deployment_cleanup_without_app_bin
+test_deployment_cleanup_without_other_docker_build_file
+
+# WARNING: This will remove all the files except "scripts"
+# create a directory called "dummy-script" then copy the entire files inside it.
+# test_deployment_cleanup_with_id_build_id
```
Comment (royge) [inline funcs_test.sh:521]: Situation like this @jarylolz12 kay sign na nga risky ang ato'ng script kay dili man siya puede ma-execute anywhere we want.

**Review by royge — CHANGES_REQUESTED** [review]

**Pull request closed without merging**